File: benchmarks/js/gc_alloc.js
class GcVtA{ constructor(x){ this.x=x; } }
class GcVtB{ constructor(y){ this.y=y; } }
function compute(){
  const junk=[]; for(let i=0;i<400000;i++) junk.push("gc_"+i);
  let aa=0,bb=0; for(let i=0;i<100000;i++){ const a=new GcVtA(i); aa+=a.x; const b=new GcVtB(i); bb+=b.y; }
  return aa+bb+junk.length;
}
for(let w=0;w<3;w++) compute();
let best=Infinity, chk;
for(let r=0;r<10;r++){ const t=performance.now(); chk=compute(); const ms=performance.now()-t; if(ms<best)best=ms; }
console.error(chk); console.log(best.toFixed(3));

File: tests/benchmarks/py/gc_alloc.py
from _harness import run


class GcVtA:
    __slots__ = ("x",)

    def __init__(self, x):
        self.x = x


class GcVtB:
    __slots__ = ("y",)

    def __init__(self, y):
        self.y = y


def compute():
    junk = []
    for i in range(400000):
        junk.append("gc_" + str(i))
    aa = 0
    bb = 0
    for i in range(100000):
        a = GcVtA(i)
        aa += a.x
        b = GcVtB(i)
        bb += b.y
    return f"check={aa}/{bb}/{len(junk)}"


run(compute)
